fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/query/plan/plans/RecordQuerySelectorPlan.java

```diff
@@ -0,0 +1,340 @@
+import com.apple.foundationdb.record.provider.foundationdb.FDBRecordStoreBase;
+import com.apple.foundationdb.record.provider.foundationdb.FDBStoreTimer;
+import com.apple.foundationdb.record.query.plan.temp.AliasMap;
+import com.apple.foundationdb.record.query.plan.temp.ExpressionRef;
+import com.apple.foundationdb.record.query.plan.temp.GroupExpressionRef;
+import com.apple.foundationdb.record.query.plan.temp.Quantifier;
+import com.apple.foundationdb.record.query.plan.temp.Quantifiers;
+import com.apple.foundationdb.record.query.plan.temp.RelationalExpression;
+import com.apple.foundationdb.record.query.plan.temp.explain.Attribute;
+import com.apple.foundationdb.record.query.plan.temp.explain.NodeInfo;
+import com.apple.foundationdb.record.query.plan.temp.explain.PlannerGraph;
+import com.apple.foundationdb.record.query.plan.temp.expressions.RelationalExpressionWithChildren;
+import com.apple.foundationdb.tuple.ByteArrayUtil2;
+import com.google.common.collect.ImmutableList;
+import com.google.protobuf.ByteString;
+import com.google.protobuf.InvalidProtocolBufferException;
+import com.google.protobuf.Message;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.Executor;
+import java.util.stream.Collectors;
+
+/**
+ * A {@link RecordQueryChooserPlanBase} that selects one of its children to be executed.
+ * This plan can be used to allow the planner to have a few optional plans to be executed where the decision on which
+ * one actually gets run is deferred to execution time.
+ * The selector can be used with a given lambda that selects among the plans, and as a convenience, a relative priority
+ * selector is also provided. The relative priority selector takes a list of priorities (each associated with a plan)
+ * and selects a plan at random with relative priority as given by the priority of the plan.
+ */
+public class RecordQuerySelectorPlan extends RecordQueryChooserPlanBase {
+    public static final Logger LOGGER = LoggerFactory.getLogger(RecordQuerySelectorPlan.class);
+    private static final ObjectPlanHash BASE_HASH = new ObjectPlanHash("Record-Query-Selector-Plan");
+
+    @Nonnull
+    private final PlanSelector planSelector;
+
+    private RecordQuerySelectorPlan(@Nonnull final List<Quantifier.Physical> quantifiers, @Nonnull final PlanSelector planSelector) {
+        super(quantifiers);
+        this.planSelector = planSelector;
+    }
+
+    /**
+     * Factory method that takes in a list of relative priorities and created a selection policy based on the priorities.
+     *
+     * @param children the list of subplans
+     * @param relativePlanPriorities a list (of the same length as the children) that determines the relative
+     * priority for selecting the plan. Sum of all the priorities must be 1.0
```

Comment:
This seems more reasonable than the actual contract here.

Reply:
Assuming the issue here is the semantics of the relative probabilities: sum of 1.0 vs. sum of 100%: This was discussed in previous conversations and modified to be % based (see below comment).